ClientUpdater/Updater.cs

```diff
@@ -15,1140 +15,1152 @@
-            List<UpdateMirror> list = new List<UpdateMirror>();
-            List<string> sectionKeys = settingsINI.GetSectionKeys("DownloadMirrors");
+        ReadUpdaterConfig();
 
-            if (sectionKeys != null)
-            {
-                foreach (string str in sectionKeys)
-                {
-                    string value = settingsINI.GetStringValue("DownloadMirrors", str, string.Empty);
-                    UpdateMirror item = updateMirrors.Find(um => um.Name == value);
-                    if ((item != null) && !list.Contains(item))
-                    {
-                        list.Add(item);
-                    }
-                }
-            }
-            foreach (UpdateMirror mirror2 in updateMirrors)
+        Logger.Log("Updater: Update mirror count: " + updateMirrors.Count);
+        Logger.Log("Updater: Running from: " + CallingExecutableFileName);
+        var list = new List<UpdateMirror>();
+        List<string> sectionKeys = settingsINI.GetSectionKeys("DownloadMirrors");
+
+        if (sectionKeys != null)
+        {
+            foreach (string str in sectionKeys)
             {
-                if (!list.Contains(mirror2))
+                string value = settingsINI.GetStringValue("DownloadMirrors", str, string.Empty);
+                UpdateMirror item = updateMirrors.Find(um => um.Name == value);
+                if (item != null && !list.Contains(item))
                 {
-                    list.Add(mirror2);
+                    list.Add(item);
                 }
             }
-            updateMirrors = list;
         }
 
-        /// <summary>
-        /// Checks if there are available updates.
-        /// </summary>
-        public static void CheckForUpdates()
+        foreach (UpdateMirror mirror2 in updateMirrors)
         {
-            Logger.Log("Updater: Checking for updates.");
-            if (VersionState != VersionState.UPDATECHECKINPROGRESS && VersionState != VersionState.UPDATEINPROGRESS)
+            if (!list.Contains(mirror2))
             {
-                new Thread(new ThreadStart(DoVersionCheck)).Start();
+                list.Add(mirror2);
             }
         }
 
-        /// <summary>
-        /// Checks version information of local files.
-        /// </summary>
-        public static void CheckLocalFileVersions()
+        updateMirrors = list;
+    }
+
+    /// <summary>
+    /// Checks if there are available updates.
+    /// </summary>
+    public static void CheckForUpdates()
+    {
+        Logger.Log("Updater: Checking for updates.");
+        if (VersionState is not VersionState.UPDATECHECKINPROGRESS and not VersionState.UPDATEINPROGRESS)
         {
-            Logger.Log("Updater: Checking local file versions.");
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+            DoVersionCheckAsync();
```

Comment:
Can't you use discard here?

Reply:
In previous reviews discards were not approved of.

Reply:
Right, let's leave it that way then, I just noticed I myself use a lot more discards @ my work project now :P

Reply:
Well, I am also in favor of using discards, it shows the explicit intent of the developer that the result was not unused by accident.